### ISSUE TITLE
Net radiation not gap-filled in LSM data

### ISSUE THREAD

**Issue by SamanthaBiegel** [geco-bern/FluxDataKit#87]
Net radiation (NETRAD variable) is only gap-filled in the rsofun drivers workflow (https://github.com/geco-bern/FluxDataKit/blob/e85d487f7ff3a0f796fea8099bf14c504a075e5f/R/helper_functions.R#L60). Is there any reason why this gap-filling step is not applied to all data products?
@stineb

**Comment by SamanthaBiegel** [comment]
Yes, I found 17% missing NETRAD values in the fluxnet-formatted data

**Comment by olivierbonte** [comment]
Hi,

First and foremost thank you @stineb and collaborators for creating this dataset, accessing an extended version of the PLUMBER2 dataset has been great so far!

For FluxDataKitv3.3, I can confirm for the sites BE-Bra, DE-Thai and DE-Tha, there is data missing for net radiation in the netcdf format. I suppose this stems from the fact that in [Ukkola et al. (2022)](https://doi.org/10.5194/essd-14-449-2022), net radiation is not considered as a meteorological variable ("Met" in Table 1 from this paper), but rather a flux, for which gapfilling is less stringent as far as I understand (as it is not used as driver for land surface models). 

For the application I am currently working on, net radiation is a forcing however, as it drivers evaporation in Priestley-Taylor/Penman-Monteith type models.

As mentioned by @SamanthaBiegel, the netrad filling is applied for formatting to rsofun drivers, see:
https://github.com/geco-bern/FluxDataKit/blob/a3aa60f101def5d2c17f89de87815fb9c1ebe687/R/fdk_format_drivers.R#L134
As far as I can see, this is not done for fluxnet formatting (basing myself on https://github.com/geco-bern/FluxDataKit/blob/24237bd46aaa4d65d9395391adf7360d53b6562b/R/fdk_process_lsm.R)
Could this also be added to the next version of FluxDataKit? Although I am not that familiar with this codebase, I would be happy to help in the implementation if need be.

Thanks in advance!
Olivier

**Comment by stineb** [comment]
Dear @olivierbonte , thanks for the suggestion. Before it is implemented (if we decide to do so), feel free to use the function `fill_netrad()` for your application. It does an imputation using K nearest neighbours. The function itself is not public (exported as part of the package), but you can source it from the source file yourself. I hope this is helpful for you.

**Comment by olivierbonte** [comment]
Thank you for the suggestion, I'll definitely consider using the function directly.
